gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/MRCompactorAvroKeyDedupJobRunner.java

```diff
@@ -99,7 +108,9 @@ protected void configureJob(Job job) throws IOException {
 
   private void configureSchema(Job job) throws IOException {
     Schema newestSchema = getNewestSchemaFromSource(job);
-    AvroJob.setInputKeySchema(job, newestSchema);
+    if (useSingleInputSchema) {
+      AvroJob.setInputKeySchema(job, newestSchema);
+    }
     AvroJob.setMapOutputKeySchema(job, this.shouldDeduplicate ? getKeySchema(job, newestSchema) : newestSchema);
```

Comment:
It is true, you still get an error in the Mapper, something like ` org.apache.avro.UnresolvedUnionException: Not in union ["null","long"]: <someStringData>` but anyway, the type of error is different and therefore solvable. Before, the error occurred while reading the data files. For my special case, I solved it by just adding some lines of code to the map() method to handle the problem (`final Object jp = key.datum().get("jp"); if (jp instanceof String) { key.datum().put("jp", Long.valueOf((String) jp));}`), for general solutions I would suggest some pluggable mechanism (e.g. sub classing `TaskInputOutputContext`) and maybe one could use version information also from AvroSchema registry, if present?

Hmm, I see you point, at first I thought it is still necessary to use `newestSchema` as  mapper output key/value schema, due to deduping, but it is true for map only jobs. 
For deduping, the error would have to be handled in the reducer, though (if the changed field is a key field). 
Maybe it is better to have one common schema for the compacted data?